matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/internal/DefaultMatchingNotificationResource.java

```diff
@@ -482,4 +491,39 @@ private Long getLastOutgoingMatchId(List<PatientMatch> matches)
         }
         return id;
     }
+
+    @Override
+    public Response computeScore(String referencePatient, String matchPatient)
+    {
+        Patient reference = createPatientFromJSON(referencePatient);
+        Patient match = createPatientFromJSON(matchPatient);
+
+        if (reference != null && match != null) {
+            try {
+                double score = this.phenotypeSimilarityScorer.getScore(reference, match);
+
+                JSONObject scoreJSON = new JSONObject();
+                scoreJSON.put("score", score);
+
+                return Response.ok(scoreJSON, MediaType.APPLICATION_JSON_TYPE).build();
+            } catch (Exception ex) {
+                this.slf4Jlogger.error("Can't compute phenotype similarity score: [{}]", ex);
+            }
+        }
+        return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+
+    private Patient createPatientFromJSON(String patientJSON)
+    {
+        JSONObject json;
+        try {
+            json = new JSONObject(patientJSON);
+        } catch (Exception ex) {
+            this.slf4Jlogger.error("Can't convert JSON to a patient: [{}]", ex.getMessage());
+            return null;
+        }
+        final Patient patient = this.repository.create();
```

Comment:
This actually creates a patient, so it is not suitable for use in production.

Some alternatives:
- Make sure you delete the patient at the end (preferably skipping the document trash), although this will create some churn
- Although using internal classes is discouraged, in this case it would be better to manually create a PhenoTipsPatient on a new XWikiDocument in a temporary space, and later delete that document
- Implement the `Patient` interface like it's done in `org.phenotips.data.script.PatientSpecificityScriptService.FakePatient`, with no actual persistence and only implementing the methods the score cares about
- Make that an actual public implementation, `TransientPatientImpl` that other services can use for temporary storage; it could extend `PhenoTipsPatient` and override methods so that they never actually save the underlying `XWikiDocument`, but otherwise it would behave just like a real patient